pkg/fanal/analyzer/language/nodejs/license.go

```diff
@@ -0,0 +1,110 @@
+package nodejs
+
+import (
+	"errors"
+	"io/fs"
+	"path"
+	"strings"
+
+	"golang.org/x/xerrors"
+
+	dio "github.com/aquasecurity/go-dep-parser/pkg/io"
+	"github.com/aquasecurity/go-dep-parser/pkg/nodejs/packagejson"
+	"github.com/aquasecurity/trivy/pkg/fanal/types"
+	"github.com/aquasecurity/trivy/pkg/licensing"
+	"github.com/aquasecurity/trivy/pkg/log"
+	"github.com/aquasecurity/trivy/pkg/utils/fsutils"
+)
+
+func ParseLicenses(
+	packageJsonParser *packagejson.Parser,
+	classifierConfidenceLevel float64,
+	licenses map[string][]string,
+) func(fsys fs.FS, root string) error {
+	return func(fsys fs.FS, root string) error {
+		if fsys == nil {
+			return xerrors.New("fs.FS required")
+		}
+
+		walkDirFunc := func(pkgJsonPath string, d fs.DirEntry, r dio.ReadSeekerAt) error {
+
+			pkg, err := packageJsonParser.Parse(r)
+			if err != nil {
+				return xerrors.Errorf("unable to parse %q: %w", pkgJsonPath, err)
+			}
+
+			ok, licenseFileName := IsLicenseRefToFile(pkg.License)
+
+			if !ok {
+				licenses[pkg.ID] = []string{pkg.License}
+				return nil
+			}
+
+			log.Logger.Debugf(
+				"Licenses are missing in %q, an attempt to find them in the LICENSE file", pkgJsonPath)
+			licenseFilePath := path.Join(path.Dir(pkgJsonPath), licenseFileName)
+
+			findings, err := classifyLicense(licenseFilePath, classifierConfidenceLevel, fsys)
+			if err != nil {
+				return err
+			}
+
+			// License found
+			if len(findings) > 0 {
+				licenses[pkg.ID] = findings.Names()
+			} else {
+				log.Logger.Debugf(
+					"The license file %q was not found or the license could not be classified", licenseFilePath)
+			}
+			return nil
+		}
+
+		if err := fsutils.WalkDir(fsys, root, isNodeModulesPkg, walkDirFunc); err != nil {
+			return xerrors.Errorf("walk error: %w", err)
+		}
+
+		return nil
+	}
+}
+
+// IsLicenseRefToFile The license field can refer to a file
+// https://docs.npmjs.com/cli/v9/configuring-npm/package-json
+func IsLicenseRefToFile(maybeLicense string) (bool, string) {
+	if maybeLicense == "" {
+		return true, "LICENSE"
```

Comment:
Can you add comment why we use `LICENSE` here?

Reply:
done